src/gluonts/torch/model/wavenet/module.py

```diff
@@ -329,24 +326,20 @@ def base_net(
 
         [Paine et al., 2016] "Fast wavenet generation algorithm."
            arXiv preprint arXiv:1611.09482 (2016).
+
         Returns
         -------
             A tensor containing the unnormalized outputs of the network of
             shape (batch_size, pred_length, num_bins) and a list containing the
             convolutional queues for each layer. The queue corresponding to
             layer `l` has shape: (batch_size, num_residual_channels, 2^l).
         """
-        if prediction_mode:
-            assert (
-                queues is not None
-            ), "Queues cannot be empty in prediction mode!"
-
         skip_outs = []
         queues_next = []
         out = inputs
         for i, layer in enumerate(self.residuals):
             skip, out = layer(out)
-            if prediction_mode:
+            if queues is not None:
```

Comment:
That's a nice simplification!

Reply:
and the fact is: type checking surfaced this, we should have it opt-out instead of opt-in